cloud/ec2.go

```diff
@@ -323,10 +323,16 @@ func (m *ec2Manager) spawnOnDemandHost(ctx context.Context, h *host.Host, ec2Set
 		ImageId:             &ec2Settings.AMI,
 		KeyName:             &ec2Settings.KeyName,
 		InstanceType:        &ec2Settings.InstanceType,
+		IamInstanceProfile:  &ec2.IamInstanceProfileSpecification{Arn: &ec2Settings.IAMInstanceProfileARN},
 		BlockDeviceMappings: blockDevices,
 		TagSpecifications:   makeTagSpecifications(makeTags(h)),
 	}
 
+	grip.Info(message.Fields{
```

Comment:
did you mean to delete this!

Reply:
yes, sorry! I converted it to a draft but it was too late lol.

Reply:
And to answer the above comment because I can't thread above: 

Yes, thank you, I need to add it to spawnSpotHost. I don't think I can add it to SpawnHost in a meaningful way though. The arn should already be in ec2 settings from c2Settings.FromDistroSettings. It then needs to be passed in to spawnOnDemandHost or spawnSpotHost.